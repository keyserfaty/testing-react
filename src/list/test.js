import React from "react";
import { create } from "react-test-renderer";
import List from ".";

describe("tests list", () => {
  const component = create(<List />);
  const rootInstance = component.root;
  const instance = component.getInstance();

  const input = rootInstance.findByType("input");
  const addButton = rootInstance.findByProps({ className: "add" });

  it("adds an item to the list", () => {
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    input.props.onChange({ target: { name: 'item', value: 'some' } });
    addButton.props.onClick();
  
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  
    const instance = component.getInstance();
    expect(instance.state.list).toContain("some");
  });
  
  it("removes an item from the list", () => {
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  
    const removeButton = rootInstance.findByProps({ className: "remove" });
    removeButton.props.onClick();
  
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    expect(instance.state.list).toEqual([]);
  });

})
