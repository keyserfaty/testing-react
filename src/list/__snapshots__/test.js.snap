// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests list adds an item to the list 1`] = `
<div>
  <input
    name="item"
    onChange={[Function]}
    value=""
  />
  <button
    className="add"
    onClick={[Function]}
  >
    + Add
  </button>
  <div
    className="list"
  >
    List: 
  </div>
</div>
`;

exports[`tests list adds an item to the list 2`] = `
<div>
  <input
    name="item"
    onChange={[Function]}
    value="some"
  />
  <button
    className="add"
    onClick={[Function]}
  >
    + Add
  </button>
  <div
    className="list"
  >
    List: 
    <div>
      <div>
        some
      </div>
      <button
        className="remove"
        onClick={[Function]}
      >
        - Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`tests list removes an item from the list 1`] = `
<div>
  <input
    name="item"
    onChange={[Function]}
    value="some"
  />
  <button
    className="add"
    onClick={[Function]}
  >
    + Add
  </button>
  <div
    className="list"
  >
    List: 
    <div>
      <div>
        some
      </div>
      <button
        className="remove"
        onClick={[Function]}
      >
        - Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`tests list removes an item from the list 2`] = `
<div>
  <input
    name="item"
    onChange={[Function]}
    value="some"
  />
  <button
    className="add"
    onClick={[Function]}
  >
    + Add
  </button>
  <div
    className="list"
  >
    List: 
  </div>
</div>
`;
